pyprophet/main.py

```diff
@@ -172,13 +171,40 @@ def save_results(self, result, extra_writes, out_pathes, pvalues):
             result.final_statistics.to_csv(full_stat_path, sep=self.delim_out, index=False)
             print "WRITTEN: ", full_stat_path
 
-        for scored_table, out_path in zip(result.scored_tables, out_pathes):
+        for input_path, scored_table, out_path in zip(self.pathes, result.scored_tables, out_pathes):
 
             cutoff = CONFIG.get("d_score.cutoff")
             scored_table.to_csv(out_path.scored_table, out_path.filtered_table, cutoff, sep=self.delim_out, index=False)
             print "WRITTEN: ", out_path.scored_table
             print "WRITTEN: ", out_path.filtered_table
 
+            if CONFIG.get("rewrite_sqmass"):
+
+                # get basepath
+                basepath = input_path.split(".tsv")[0]
+                basepath = basepath.split(".txt")[0]
+                basepath = basepath.split(".csv")[0]
+
+                # try to find a matching sqMass file
+                sqmass_file = None
+                if os.path.exists(basepath + ".chrom.sqMass"):
+                    sqmass_file = basepath + ".chrom.sqMass"
+                elif os.path.exists(basepath + ".sqMass"):
+                    sqmass_file = basepath + ".sqMass"
+
+                # get selected chromatograms on the filtered table
+                df = scored_table.df[scored_table.df.d_score > cutoff]
+                fragment_anno = df.aggr_Fragment_Annotation.unique()
+                prec_anno = df.aggr_prec_Fragment_Annotation.unique()
```

Comment:
@grosenberger can you comment on whether/how it would be possible to get the fragment/precursor labels from the SQL database? that would be critical for this to work with .osw sqlite files? I guess we would need `TRANSITION_ID` from those tables?